tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -51,6 +55,55 @@ bool GetEnvBool(const char* name, bool defval) {
 
 }  // namespace
 
+// This function gets pid of a process and checks if that process is using tpu.
+// It is not able to check processes that are owned by another user.
+bool IsTpuUsed(long pid) {
+  std::string path = absl::StrCat("/proc/", pid, "/fd");
+  DIR* raw_fd_dir = opendir(path.c_str());
+  if (!raw_fd_dir) {
+    return false;
+  }
+  std::unique_ptr<DIR, int (*)(DIR*)> fd_dir(raw_fd_dir, closedir);
+  struct dirent* ent;
+  char line[100];
+  while (ent = readdir(raw_fd_dir)) {
+    if (!isdigit(*ent->d_name)) continue;
+    long fd = strtol(ent->d_name, NULL, 10);
+    path = absl::StrCat("/proc/", pid, "/fd/", fd);
+    if (!readlink(path.c_str(), line, 100)) return false;
```

Comment:
Should this be `continue` instead of `return false`?

Reply:
I thought if you are not able to read one of them, that means you are unable to read all of them but I am not sure! so I changed it to continue as you suggested.